cli/cmd/service.go

```diff
@@ -1957,3 +1977,85 @@ func (c *ServicedCli) cmdServiceClearEmergency(ctx *cli.Context) {
 
 	fmt.Printf("Cleared emergency status for %d services\n", count)
 }
+
+// serviced service tune SERVICEID
+func (c *ServicedCli) cmdServiceTune(ctx *cli.Context) {
+	args := ctx.Args()
+	if len(args) < 1 {
+		fmt.Printf("Incorrect Usage.\n\n")
+		cli.ShowCommandHelp(ctx, "tune")
+		return
+	}
+
+	svcDetails, _, err := c.searchForService(args[0])
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		return
+	}
+
+	service, err := c.driver.GetService(svcDetails.ID)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		return
+	}
+
+	// Check the arguments
+	modified := false
+	if ctx.IsSet("instances") {
+		oldInstanceCount := service.Instances
+		newInstanceCount := ctx.Int("instances")
+		if oldInstanceCount != newInstanceCount {
+			service.Instances = newInstanceCount
+			modified = true
+		}
+	}
+
+	if ctx.IsSet("ramCommitment") {
+		oldCommitment := service.RAMCommitment
+		ramCommitment, err := utils.ParseEngineeringNotation(ctx.String("ramCommitment"))
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return
+		}
+		newCommitment := utils.NewEngNotation(int64(ramCommitment))
+
+		if oldCommitment.Value != newCommitment.Value {
+			service.RAMCommitment = newCommitment
+			modified = true
+		}
+	}
+
+	if ctx.IsSet("ramThreshold") {
+		oldThreshold := service.RAMThreshold
+		ramThreshold := ctx.String("ramThreshold")
+		newThreshold, err := utils.ParsePercentage(ramThreshold, service.RAMCommitment.Value)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return
+		}
+		if uint64(oldThreshold) != newThreshold {
+			service.RAMThreshold = uint(newThreshold)
+			modified = true
+		}
+	}
+
+	if modified {
+		jsonService, err := json.MarshalIndent(service, " ", "  ")
```

Comment:
This seems a little round-about to me. We already have a `Service` object, and then we turn it into JSON, to pass to `c.driver.UpdateService()` which turns the JSON back into a `Service` object  I think it would be cleaner to avoid the JSON back-and-forth.

You could add something like `UpdateServiceObj()` to `cli/api/service.go`, call it here and refactor to `UpdateService()` in `cli/api/service.go` to call it as well

Reply:
Can do. I was going for low wake; I'll add the new function.